### PR TITLE
feat(sql): add first and last boolean aggregation functions

### PR DESCRIPTION
**Pull request by rexf**

### CODE REVIEW

File: core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstBooleanGroupByFunctionFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.junit.Test;
+
+public class FirstBooleanGroupByFunctionFactoryTest extends AbstractCairoTest {
+
+    @Test
+    public void testAllNull() throws SqlException {
+
+        ddl("create table tab (f boolean)");
+
+        try (TableWriter w = getWriter("tab")) {
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstBooleanGroupByFunctionFactoryTest.java:43]: nit: there is no need to deal with table writers and record cursor factories in these tests. You could simply use our utility methods like `assertQuery`:
https://github.com/questdb/questdb/blob/e4b6520e7b33187a0d752d53748aa7e981db3f81/core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstSymbolGroupByFunctionFactoryTest.java#L33-L45

File: core/src/main/java/io/questdb/griffin/engine/functions/BooleanFunction.java [hunk 2]
```diff
@@ -179,6 +179,11 @@ public final int getType() {
         return ColumnType.BOOLEAN;
     }
 
+    @Override
+    public boolean getBool(Record rec) {
+        return getBool(rec);
```
[the remark this answers is not in the document]
Comment (rexf) [reply core/src/main/java/io/questdb/griffin/engine/functions/BooleanFunction.java]: nice catch. removed.

File: core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstBooleanGroupByFunctionFactoryTest.java (hunk 1 above)
Comment (rexf) [reply core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstBooleanGroupByFunctionFactoryTest.java:43]: converted to assertQuery,

Comment (rexf) [reply core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstBooleanGroupByFunctionFactoryTest.java:43]: Still keeping the `testAllNull` using TableWriter.
Boolean does not have a null.

**Review by puzpuzpuz — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**